src/codegen_fn_generic.cpp

```
@@ -0,0 +1,444 @@
+        return "str";
+    }
+
+    if (auto *st = llvm::dyn_cast<llvm::StructType>(ty)) {
+        std::string n = findStructTypeName(st);
+        if (!n.empty()) return n;
+    }
+
+    return "any";
+}
+
+std::vector<std::string> CodeGen::inferTypeArgs(
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)